tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/generalize_zip_codes_test.py

```diff
@@ -0,0 +1,143 @@
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+        super().initialize_class_vars()
+
+        # set the test project identifier
+        project_id = os.environ.get(PROJECT_ID)
+        cls.project_id = project_id
+
+        # set the expected test datasets
+        dataset_id = os.environ.get('COMBINED_DATASET_ID')
+        cls.dataset_id = dataset_id
+        sandbox_id = dataset_id + '_sandbox'
+        cls.sandbox_id = sandbox_id
+
+        cls.rule_instance = GeneralizeZipCodes(project_id, dataset_id,
+                                               sandbox_id)
+
+        cls.fq_table_names = [
+            f'{project_id}.{dataset_id}.{OBSERVATION}',
+        ]
+
+        # call super to set up the client, create datasets, and create
+        # empty test tables
+        # NOTE:  does not create empty sandbox tables.
+        super().setUpClass()
+
+    def setUp(self):
+        fq_dataset_name = self.fq_table_names[0].split('.')
+        self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
+
+        super().setUp()
+
+    def test_generalize_zip_codes_cleaning(self):
+        """
+        Tests that the sepcifications for GENERALIZE_ZIP_CODES_QUERY perform as designed.
+
+        Validates pre conditions, tests execution, and post conditions based on the load
+        statements and the tables_and_counts variable.        
+        """
+        queries = []
+
+        #Append some queries
+        zipcodes_tmpl = self.jinja_env.from_string("""
+            INSERT INTO `{{fq_dataset_name}}.observation`
+            (observation_id, person_id, observation_concept_id, observation_date, 
+             observation_type_concept_id, observation_source_value, observation_source_concept_id,
+             value_as_string)
+             VALUES
+                (1001, 1001, 0, '2020-08-30', 1001, 'StreetAddress_PIIZIP', 1585250, '60512'),
+                (1002, 1002, 0, '2020-08-30', 1002, 'StreetAddress_PIIZIP', 1585250, '10832'),
+                (1003, 1003, 0, '2020-08-30', 1003, 'StreetAddress_PIIZIP', 1585250, '97261'),
+                (1004, 1004, 0, '2020-08-30', 1004, 'StreetAddress_PIIZIP', 1585250, '93431'),
+                (1005, 1005, 0, '2020-08-30', 1005, 'StreetAddress_PIIZIP', 1585250, '00589'),
+                (1006, 1006, 0, '2020-08-30', 1006, 'StreetAddress_PIIZIP', 1585250, '23591'),
+                (1007, 1007, 0, '2020-08-30', 1007, 'StreetAddress_PIIZIP', 1585250, '23512-4'),
+                (1008, 1008, 0, '2020-08-30', 1008, 'StreetAddress_PIICity', 1585248, 'New York'),
+                (1009, 1009, 0, '2020-08-30', 1009, 'StreetAddress_PIIState', 1585249, 'NY'),
+                (1010, 1010, 0, '2020-08-30', 1010, 'StreetAddress_PIIZIP', 1585250, '72321')
+        """).render(fq_dataset_name=self.fq_dataset_name)
```

Comment:
Added another record to make sure that integers that don't have 1585250 aren't obscured.
```suggestion
                (1010, 1010, 0, '2020-08-30', 1010, 'StreetAddress_PIIZIP', 1585250, '72321'),
                (1011, 1011, 0, '2020-08-30', 1011, 'StreetAddress_PIIZIP', 1585247, '37218')
        """).render(fq_dataset_name=self.fq_dataset_name)
```